File: src/main/java/org/example/iotbay/service/OrderService.java
package org.example.iotbay.service;

import org.example.iotbay.domain.Order;
import org.example.iotbay.dto.OrderDTO;
import org.example.iotbay.dto.OrderDTO.Request;
import org.example.iotbay.dto.OrderDTO.Response;

import java.util.List;

public interface OrderService {
    Response createOrder(Request request);

    Order getOrderById(Long orderId);
    List<Order> getOrdersByUserId(Long userId);
    Response updateOrder(Long orderId, Request request);
    String deleteOrder(Long orderId);

}

File: src/services/OrderService.js
import axios from "axios";

class OrderService {
    BASE_URL = `http://localhost:8080/api`;

    createOrder(orderData) {
        if (!orderData.userId) {
            console.error("User ID is undefined");
            return Promise.reject(new Error("User ID is undefined"));
        }
        return axios.post(`${this.BASE_URL}/order/create`, orderData)
            .then(response => response.data)
            .catch(error => {
                console.error('Failed to create order:', error);
                throw error;
            });
    }

    deleteOrder(orderId) {
        return axios.delete(`${this.BASE_URL}/order/${orderId}`);
    }

    updateOrder(orderId, orderData) {
        return axios.put(`${this.BASE_URL}/order/${orderId}`, orderData)
            .then(response => response.data)
            .catch(error => {
                console.error('Failed to update order:', error);
                throw error;
            });
    }

    getOrderById(orderId) {
        return axios.get(`${this.BASE_URL}/order/${orderId}`)
            .then(response => response.data)
            .catch(error => {
                console.error('Failed to get order:', error);
                throw error;
            });
    }

    listOrdersByUser(userId) {
        return axios.get(`${this.BASE_URL}/order/user/${userId}`)
            .then(response => response.data)
            .catch(error => {
                console.error('Failed to list orders:', error);
                throw error;
            });
    }
}

const orderService = new OrderService();
export default orderService;
